share/completions/dnf.fish

```diff
@@ -2,4 +2,78 @@
 # Completions for the dnf command
 #
 
-complete -c dnf -w yum
+# All dnf commands
+
+# Test if completing using package names is appropriate
+function __fish_dnf_package_installed
```

Comment:
This entire function (and the next) is just `__fish_seen_subcommand_from`, e.g. `__fish_seen_subcommand_from update upgrade remove erase reinstall`. Please use that.